src/titiler/xarray/titiler/xarray/io.py

```diff
@@ -1,20 +1,40 @@
 """titiler.xarray.io"""
 
+from __future__ import annotations
+
+import os
+import re
+from functools import cache
+from pathlib import Path
 from typing import Any, Callable, Dict, List, Literal, Optional, Union
 from urllib.parse import urlparse
 
 import attr
+import httpx
+import obstore
 import xarray
+import zarr
 from morecantile import TileMatrixSet
 from rio_tiler.constants import WEB_MERCATOR_TMS
 from rio_tiler.io.xarray import XarrayReader
-from xarray.namedarray.utils import module_available
+from zarr.storage import ObjectStore
+
 
+def _find_bucket_region(bucket: str, use_https: bool = True) -> str | None:
+    prefix = "https" if use_https else "http"
+    response = httpx.get(f"{prefix}://{bucket}.s3.amazonaws.com")
+    return response.headers.get("x-amz-bucket-region")
 
-def xarray_open_dataset(  # noqa: C901
+
+@cache
+def open_zarr(  # noqa: C901
     src_path: str,
     group: Optional[str] = None,
     decode_times: bool = True,
+    decode_coords: str = "all",
+    consolidated: bool = True,
```

Comment:
xarray will try to use consolidated metadata by default, and fall back to non-consolidated if it doesn't exist. By setting the default to consolidated=True, you will tell xarray to raise an error if consolidated metadata doesn't exist rather than falling back to distributed metadata. I'm not sure this restriction is worthwhile.

Reply:
🙏 makes sense